camel-core/src/test/java/org/apache/camel/ComponentVerifierTest.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel;
+import junit.framework.TestCase;
+import org.apache.camel.ComponentVerifier.VerificationError;
+import org.apache.camel.impl.verifier.ResultErrorBuilder;
+import org.junit.Assert;
+
+public class ComponentVerifierTest extends TestCase {
+
+    public void testGetErrorDetails() {
+        VerificationError error =
+            ResultErrorBuilder.withCodeAndDescription(VerificationError.asCode("test_code"), "test error desc")
+                              .detail(VerificationError.asAttribute("test_attr_1"), "test_detail_1")
+                              .detail(VerificationError.HttpAttribute.HTTP_CODE,"test_detail_2")
+                              .detail("HTTP_TEXT", "test_detail_3")
+                              .build();
+        Assert.assertEquals("test_detail_1", error.getDetail(VerificationError.asAttribute("test_attr_1")));
+        Assert.assertEquals("test_detail_1", error.getDetail("test_attr_1"));
+        Assert.assertEquals("test_detail_2", error.getDetail(VerificationError.HttpAttribute.HTTP_CODE));
+
+        // Standard codes and attributes can be only obtained via their constants
+        Assert.assertNull(error.getDetail(VerificationError.asAttribute("HTTP_CODE")));
+        Assert.assertNull(error.getDetail("HTTP_CODE"));
+
```

Comment:
Thinking aloud, should we eventually remove enums and have just interfaces ? 
Something like:

```java
interface HttpAttribute extends Attribute {
    static HttpAttribute HTTP_CODE = HttpAttribute.asAttribute("HTTP_CODE");

    static HttpAttribute asAttribute(String name) {
        ...
    }
}
```

So that we won't have differences and instanceof is still an option ?